plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHivePartitionedBucketFunction.java

```diff
@@ -0,0 +1,151 @@
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.block.BlockAssertions.createLongRepeatBlock;
+import static io.trino.block.BlockAssertions.createLongSequenceBlock;
+import static io.trino.block.BlockAssertions.createLongsBlock;
+import static io.trino.plugin.hive.HiveType.HIVE_LONG;
+import static io.trino.plugin.hive.util.HiveBucketing.BucketingVersion;
+import static io.trino.plugin.hive.util.HiveBucketing.BucketingVersion.BUCKETING_V1;
+import static io.trino.plugin.hive.util.HiveBucketing.BucketingVersion.BUCKETING_V2;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static org.testng.Assert.assertEquals;
+
+public class TestHivePartitionedBucketFunction
+{
+    @DataProvider(name = "hiveBucketingVersion")
+    public static Object[][] hiveBucketingVersion()
+    {
+        return new Object[][] {{BUCKETING_V1}, {BUCKETING_V2}};
+    }
+
+    @Test(dataProvider = "hiveBucketingVersion")
+    public void testSinglePartition(BucketingVersion hiveBucketingVersion)
+    {
+        int numValues = 1024;
+        int numBuckets = 10;
+        Block bucketColumn = createLongSequenceBlock(923402935, 923402935 + numValues);
+        Page bucketedColumnPage = new Page(bucketColumn);
+        Block partitionColumn = createLongRepeatBlock(78758, numValues);
+        Page page = new Page(bucketColumn, partitionColumn);
+        BucketFunction hiveBucketFunction = bucketFunction(hiveBucketingVersion, numBuckets, ImmutableList.of(HIVE_LONG));
+        Multimap<Integer, Integer> bucketPositions = HashMultimap.create();
+
+        for (int i = 0; i < numValues; i++) {
+            int hiveBucket = hiveBucketFunction.getBucket(bucketedColumnPage, i);
+            // record list of positions for each hive bucket
+            bucketPositions.put(hiveBucket, i);
+        }
+
+        BucketFunction hivePartitionedBucketFunction = partitionedBucketFunction(hiveBucketingVersion, numBuckets, ImmutableList.of(HIVE_LONG), ImmutableList.of(BIGINT), 100);
+        // All positions of a bucket should hash to the same partitioned bucket
+        for (Map.Entry<Integer, Collection<Integer>> entry : bucketPositions.asMap().entrySet()) {
+            assertBucketCount(hivePartitionedBucketFunction, page, entry.getValue(), 1);
+        }
+
+        assertBucketCount(
+                hivePartitionedBucketFunction,
+                page,
+                IntStream.range(0, numValues).boxed().collect(toImmutableList()),
+                numBuckets);
+    }
+
+    @Test(dataProvider = "hiveBucketingVersion")
+    public void testMultiplePartitions(BucketingVersion hiveBucketingVersion)
+    {
+        int numValues = 1024;
+        int numBuckets = 10;
+        Block bucketColumn = createLongSequenceBlock(923402935, 923402935 + numValues);
```

Comment:
could you extract this to separate method and append a null value also?

Reply:
done